doc/manual/src/command-ref/files/profiles.md

```diff
@@ -1,17 +1,42 @@
-## Profiles
+## Profile Directory
 
-A directory that contains links to profiles managed by [`nix-env`] and [`nix profile`]:
+A directory that contains profiles.
+Profiles are typically managed by [`nix-env`] and [`nix profile`].
+These are the default profile directories:
 
 - `$XDG_STATE_HOME/nix/profiles` for regular users
 - `$NIX_STATE_DIR/profiles/per-user/root` if the user is `root`
 
-A profile is a directory of symlinks to files in the Nix store.
+## Profile
+
+Profiles are a basic version control mechanism built atop more core Nix constructs and symlinks.
+Profiles consiste of
+
+- An map of zero or more generations, each mapping a *generation number* to a [store object](@docroot@/glossary.md#gloss-store-object).
+
+- An optional *current* or *active* generation, which is one of those in the map.
+
```

Comment:
This definition is precise but doesn't really explain what profiles are for, other than a "basic version control mechanism". It should probably have some introductory line that explains that profiles enable atomic upgrades and rollbacks of (sets of) packages.

Reply:
This is the "files" section I thought was supposed to be more about how things are represented on disk. I think separately the concepts / data model itself should be documented without this minutiae.

(C.f. Platonic "file system object"s vs the grubby details of normalizing timestamps, permissions, file system ACLs, and other such things.)